### PR TITLE
Search and address management

### PR DESCRIPTION
**Pull request by sanu29**
Added Profile page
image

Added Address API and Address Page

image

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -69,4 +75,59 @@ body {
...
\ No newline at end of file
+}
+.search-button{
+    flex-shrink: 0;
+}
+
+
+.options{
+    height: fit-content;
+    width: 5rem  !important;
```
Comment (sanjitsarkar) [inline src/App.css]: Don't use important attribute.

File: src/App.css [hunk 2]
```diff
@@ -69,4 +75,59 @@ body {
...
+    top: 3rem;
+    right: 0rem;
+    justify-content: center;
+    align-items: center;
+    padding: 0.5rem 0;
+    box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
```
Comment (sanjitsarkar) [inline src/App.css:92]: Use variables for color

File: src/App.js [hunk 3]
```diff
@@ -21,6 +23,9 @@ function App() {
             <Route element={<PrivateRoute/>}>
             <Route path="/wishlist" element={<Wishlist/>}/>
             <Route path="/cart" element={<Cart/>}/>
+            <Route path="/Profile" element={<Profile/>}/>
+            <Route path="/Address" element={<Address/>}/>
+            {/* <Route path="/Profile" element={<Profile/>}/> */}
```
Comment (sanjitsarkar) [inline src/App.js]: Remove commented code

File: src/component/AddressBody.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import "../pages/auth.css"
+import { AddressForm } from "./AddressForm";
+export function AddressBody() {
+
+
+
+    const [addNew, setAddNew] = useState("none")
+    const {pincodeDes} = UseAddressContext()
+    const {addAddress, address , removeAddress} = UseAddressContext();
+    const ListOfAddress = () =>{
```
Comment (sanjitsarkar) [inline src/component/AddressBody.jsx:13]: No need to check for [], undefined
Just check it for value it returns when there's no data.

File: src/component/AddressBody.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+         if(address === [] || address === "" || address === undefined|| address.length <1)
```
Comment (sanjitsarkar) [inline src/component/AddressBody.jsx:14]: Array length cannot smaller than zero and arr===[] is similar to arr.length===0

File: src/component/AddressForm.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { UseAddressContext } from "../context/address-context";
+
+export function AddressForm(disp,setDisp) {
+
+    const [name, setname] = useState("");
+    const [mobileNumber, setmobileNumber] = useState("");
+    const [pincode, setpinCode] = useState("");
+    const [Location, setLocation] = useState("");
```
Comment (sanjitsarkar) [inline src/component/AddressForm.jsx]: Name the useState variable with cameCase

File: src/component/AddressForm.jsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    const [city, setcity] = useState("");
+    const [Address, setAddress] = useState("");
+    const [state, setstate] = useState("");
```
Comment (sanjitsarkar) [inline src/component/AddressForm.jsx:11]: You can separate those setError,state to a context like AddressmanagementContext

File: src/component/AddressForm.jsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                    onBlur={(e) => e.target.value === "" || e.target.value.length !== 6 ? seterror("Pincode format incorrect") : ""}
+                    value={pincode} />
+            </div>
+
+        </div>
+        <div className="address-form d-flex justify-content-center  w-100 flex-wrap w-100 ">
+            <div className="d-flex-column w-100">
+                <label htmlFor="Name" className="para5 w-100  m-8 label">Address</label>
+                <textarea className="input border-color-gray  w-100"
+                    onChange={(e) => setAddress(e.target.value)}
```
Comment (sanjitsarkar) [inline src/component/AddressForm.jsx:51]: No need

**Review by sanjitsarkar — APPROVED** [review]
UI looks simple and good looking.
Rest of the codes look okay
Looks Good to merge.

**Review by srishtigautam21 — APPROVED** [review]
The code looks clean and understandable
good to merge

**Pull request merged**